client_plugin/drivers/vfile/vfile_driver.go

```diff
@@ -562,6 +613,11 @@ func (d *VolumeDriver) MountVolume(name string, fstype string, id string, isRead
 		if err != nil {
```

Comment:
As we discussed, it is still possible that the mount volume could fail and the atomic decrement and removeVMFromClientList fails. In such a case, the data in etcd will say that the volume is successfully attached, whereas in reality that has not happened.

I though more on this, could we do this instead?
- Mark the volume to be mounted as 'mounting' to node
- Mount the volume
- If mount is success, mark the volume as 'mounted' and increment the global ref count.
- If failure, try to undo the above operations in kv. If this fails, we just log the error.

We can have a background task that continuously keeps monitoring volumes that are in 'mounting' state, and mark as failure after some timeout.

Would this approach work?

Reply:
No this won't work...
The reason is, "increase global ref count" has to happen before "Mount the volume".
This is because global ref count change triggered the swarm managers to start the mounting process.